src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx

```diff
@@ -32,69 +32,74 @@ import HealthCareGroupSupportingText from './HealthCareGroupSupportingText';
   mobilePhoneNumber,
-  noContactInfoOnFile,
-  notificationGroups,
-  shouldFetchNotificationSettings,
   shouldShowAPIError,
   shouldShowLoadingIndicator,
 }) => {
   const location = useLocation();
 
-  const { TOGGLE_NAMES, useToggleValue } = useFeatureToggle();
-  const showQuickSubmitGroup = useToggleValue(
-    TOGGLE_NAMES.profileShowQuickSubmitNotificationSetting,
-  );
-
-  React.useEffect(() => {
-    // issue: 48011
-    // used via passed state from contact info - mobile update alert link
-    if (location.state?.scrollToTop) {
-      scroll.scrollToTop({ duration: 0, smooth: false });
-    }
-
-    focusElement('[data-focus-target]');
-    document.title = `Notification Settings | Veterans Affairs`;
-  }, []);
+  const {
+    toggles: notificationToggles,
+    useAvailableGroups,
+  } = useNotificationSettingsUtils();
 
-  React.useEffect(
+  const requiredContactInfoOnFile = useMemo(
     () => {
-      if (shouldFetchNotificationSettings) {
-        fetchNotificationSettings({ facilities });
-      }
+      return notificationToggles?.showEmailNotificationSettings
+        ? !!(emailAddress || mobilePhoneNumber)
+        : !!mobilePhoneNumber;
     },
-    [fetchNotificationSettings, shouldFetchNotificationSettings],
+    [emailAddress, mobilePhoneNumber, notificationToggles],
   );
 
-  // if either phone number or email address is not set
-  const showMissingContactInfoAlert = React.useMemo(
+  const showMissingContactInfoAlert = useMemo(
     () =>
       !shouldShowLoadingIndicator &&
       !shouldShowAPIError &&
-      !allContactInfoOnFile,
-    [allContactInfoOnFile, shouldShowAPIError, shouldShowLoadingIndicator],
+      !requiredContactInfoOnFile,
+    [requiredContactInfoOnFile, shouldShowAPIError, shouldShowLoadingIndicator],
   );
 
-  // shown as long as we aren't loading data and they have at least one
-  // communication channel on file
-  const showNotificationOptions = React.useMemo(
+  const shouldFetchNotificationSettings = useMemo(
     () => {
-      return (
-        !shouldShowLoadingIndicator &&
-        !shouldShowAPIError &&
-        !noContactInfoOnFile
-      );
+      return !showMissingContactInfoAlert && !shouldShowAPIError;
+    },
+    [showMissingContactInfoAlert, shouldShowAPIError],
+  );
+
+  useEffect(
+    () => {
+      // issue: 48011
+      // used via passed state from contact info - mobile update alert link
+      if (location.state?.scrollToTop) {
+        scroll.scrollToTop({ duration: 0, smooth: false });
+      }
+
+      focusElement('[data-focus-target]');
+      document.title = `Notification Settings | Veterans Affairs`;
     },
-    [noContactInfoOnFile, shouldShowAPIError, shouldShowLoadingIndicator],
+    [location.state?.scrollToTop],
   );
 
+  useEffect(
+    () => {
+      if (shouldFetchNotificationSettings) {
+        fetchNotificationSettings({
+          facilities,
+        });
+      }
+    },
+    [fetchNotificationSettings, shouldFetchNotificationSettings],
```

Comment:
So this eslint warning... if I add `facilities` to the array of deps, I get a dreaded infinite loop. Not sure how I should fix that. 

Maybe a `useMemo` on the facilities alone at the top of the component would fix it, or just leave it as is? I don't think it will cause an issue but 🤷🏼